Sources/KlaviyoSwift/APIRequestErrorHandling.swift

```diff
@@ -26,7 +62,13 @@ func handleRequestError(
     case let .httpError(statuscode, data):
         let responseString = String(data: data, encoding: .utf8) ?? "[Unknown]"
         environment.logger.error("An http error occured status code: \(statuscode) data: \(responseString)")
-        return .dequeCompletedResults(request)
+
+        let invalidFields = parseError(data)
+        if let invalidFields, !invalidFields.isEmpty {
+            return .resetStateAndDequeue(request, invalidFields)
+        } else {
+            return .dequeCompletedResults(request)
+        }
```

Comment:
Trivial/spelling: I've noticed we used both `dequeue` and `deque` spellings. I'd advocate for picking a consistent spelling, and also camel-casing it like `deQueue` to distinguish from `deque`/`dequeue` which is a type of queue with two ends ([double-ended queue](https://www.geeksforgeeks.org/difference-between-queue-and-deque-queue-vs-deque/)).

Reply:
Good catch! and thanks for the suggestion. Using camelCase and In the interesting of keeping this PR readable I created a separate PR with just the rename changes [here](https://github.com/klaviyo/klaviyo-swift-sdk/pull/128). Let me know if it looks good.